python-mini-projects/projects/Merge_pdfs/merge_pdfs.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
+
+from PyPDF2 import PdfFileMerger
+
+
+# By appending in the end
+def by_appending():
+    merger = PdfFileMerger()
+    # Either provide file stream
+    f1 = open("samplePdf1.pdf", "rb")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *f1*. The resource is allocated by call *builtins.open*. Currently, there are execution paths that do not contain closure statements, for example, when *append()* throws an exception. To prevent this resource leak, close *f1* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).